frappe/core/doctype/data_import/importer.py

```diff
@@ -664,6 +664,20 @@ def validate_value(self, value, col):
 					}
 				)
 				return
+		elif df.fieldtype == "Duration":
+			import re
+			is_valid_duration = re.match("^(?:(\d+d)?((^|\s)\d+h)?((^|\s)\d+m)?((^|\s)\d+s)?)$", value)
```

Comment:
It will be better if this is checked at the `document.py` level

Reply:
The `duration_to_seconds` function is called while setting value for the field. Since at the `document.py` level the value will be in seconds, it has to be checked during the conversion. Added that. However, it is okay to show a warning before import too?

Reply:
I mean the warning should come from `document.py`, but this is also fine I guess.